src/main/java/fr/openmc/core/features/dream/generation/structures/DreamStructure.java

```diff
@@ -0,0 +1,93 @@
+package fr.openmc.core.features.dream.generation.structures;
+
+import com.sk89q.worldedit.math.BlockVector3;
+import fr.openmc.core.features.dream.DreamUtils;
+import lombok.Getter;
+import org.bukkit.Location;
+
+import java.util.Objects;
+
+public record DreamStructure(DreamType type, BlockVector3 min, BlockVector3 max) {
+
+    public boolean isInside(Location loc) {
+        if (!DreamUtils.isDreamWorld(loc)) return false;
+
+        double x = loc.getX();
+        double y = loc.getY();
+        double z = loc.getZ();
+
+        double minX = Math.min(min.x(), max.x());
+        double maxX = Math.max(min.x(), max.x());
+        double minY = Math.min(min.y(), max.y());
+        double maxY = Math.max(min.y(), max.y());
+        double minZ = Math.min(min.z(), max.z());
+        double maxZ = Math.max(min.z(), max.z());
+
+        return x >= minX && x <= maxX
+                && y >= minY && y <= maxY
+                && z >= minZ && z <= maxZ;
+    }
+
+    public static DreamStructure fromString(String s) {
```

Comment:
```suggestion
    public static DreamStructure deserialize(String s) {
```
Regarde ConfigurationSerialization